lib/Dialect/FIRRTL/FIRRTLAnnotationHelper.cpp

```diff
@@ -777,6 +777,65 @@ LogicalResult circt::firrtl::applyGCTMemTaps(const AnnoPathValue &target,
   }
 
   ImplicitLocOpBuilder builder(combMem->getLoc(), combMem);
+
+  // Lower memory taps to real ports on the memory.  This is done if the taps
+  // are supposed to be synthesized.
+  if (state.noRefTypePorts) {
+    // Create new ports _after_ all other ports to avoid permuting existing
+    // ports.
+    builder.setInsertionPointToEnd(
```

Comment:
Probably want this to insert at end of the block containing the memory?

This modification to the test case causes this to produce invalid IR:

```
  module DUTModule :
    input clock : Clock
    input reset : Reset
    output io : { flip addr : UInt<3>, flip dataIn : UInt<8>, flip wen : UInt<1>, dataOut : UInt<8>}

    io.dataOut is invalid
    when UInt<1>(1) :
      cmem rf : UInt<8> [8]
      infer mport read = rf[io.addr], clock
      io.dataOut <= read
      when io.wen :
        infer mport write = rf[io.addr], clock
        write <= io.dataIn
```